docs/release_process.md

```diff
@@ -0,0 +1,319 @@
+long as it can be done with a fast-forward push. Since the
+[Great GitHub Heroku Incident of 2022][github-heroku-incident], this is
+done from CircleCI using a [service account][service-account].
+
+To push a different branch, you need to add the Heroku app as a remote:
+
+* `heroku login`
+* `heroku git:remote -a fx-breach-alerts`
+
+Then, you can push your local unmerged branch to Heroku:
+
+* `git push -f heroku change-1:main`
+
+Merges to main will fail to deploy until someone manually resets it to `main`:
+
+* `git push -f heroku main`
+
+
+## Release to Stage
+Every tag pushed to GitHub is automatically deployed to the [Stage][stage]
+server. The standard practice is to create a tag from `main` every Tuesday at
+the end of the day, and to name the tag with `YYYY-MM-DD` [CalVer][calver]
+syntax. This tag will include only the changes that have been merged to `main`.
+E.g.,
+
+1. `git tag 2022.08.02`
+2. `git push --tags`
+
+E.g., the following `2022.08.02` tag includes only `change-1` and `change-2`.
+```mermaid
+%%{init: { 'theme': 'base', 'gitGraph': {'rotateCommitLabel': true} } }%%
+    gitGraph
+       commit
+       branch change-1
+       commit
+       commit
+       checkout main
+       branch change-2
+       commit
+       checkout main
+       merge change-1
+       branch change-3
+       commit
+       commit
+       checkout main
+       merge change-2 tag: "2022.08.02"
+       checkout change-3
+       commit
+       commit
+       checkout main
+```
+
+### Create Release Notes on GitHub
+After you push the tag to GitHub, you should also
+[make a pre-release on GitHub][github-new-release] for the tag.
+
+1. Choose the tag you just pushed (e.g., `2022.08.02`)
+2. Type the same tag name for the releae title (e.g., `2022.08.02`)
+3. Click "Previous tag:" and choose the tag currently on production.
+   * You can find this at [the `__version__` endpoint][prod-version].
+4. Click the "Generate release notes" button!
+5. Check the pre-release box.
+6. Click "Publish release"
+
+
+## Release to Prod
+We leave the tag on [Stage][stage] for a week so that we (and especially QA)
+can check the tag on GCP infrastucture before we deploy it to production. To
+deploy the tag to production:
+
+1. File an [SRE ticket][sre-board] to deploy the tag to [Prod][prod].
+   * Include a link to the GitHub Release
+   * You can assign it directly to our primary SRE for the day
+2. When SRE starts the deploy, "cloudops-jenkins" will send status messages
+   into the #fx-monitor-engineering channel.
+3. When you see `PROMOTE PROD COMPLETE`, do some checks on prod:
+   * Check sentry prod project for a spike in any new issues
+   * Check grafana dashboard for any unexpected spike in ops
+   * Spot-check the site for basic functionality
+   * Ping SDET to run end-to-end tests on prod
+4. Update the GitHub Release from "pre-release" to a full release and reference the production deploy SRE Jira ticket.
```

Comment:
Hmm, who is that for? Since GH Releases are publicly-viewable, whereas I assume the SRE ticket isn't.

Or is it the other way around, link to the release in the SRE ticket?

Reply:
I guess for us, or anyone that wants to have an authoritative place for folks to check to see what the latest release is, I find it convenient when looking at code and trying to figure out what's shipped and what's not.

Reply:
Just in case it's useful: if you `git blame` a line and view that commit in GitHub ([example](https://github.com/mozilla/blurts-server/commit/db9b115a82312a0b5a8602a7fc7a3b080cf77802)), then it will list which tags include that commit:

![image](https://github.com/mozilla/blurts-server/assets/4251/43aa0580-92d0-41fb-a5d3-5c8509f621f3)

I've often used this (combined with the `__version__` endpoint) to figure out where something has or has not shipped yet.